clients/git/client.go

```diff
@@ -207,3 +261,84 @@ func (c *Client) Close() error {
 	}
 	return nil
 }
+
+func (c *Client) GetBranch(branch string) (*clients.BranchRef, error) {
+	// Get the branch reference
+	ref, err := c.gitRepo.Branch(branch)
+	if err != nil {
+		return nil, fmt.Errorf("git.Branch: %w", err)
+	}
+
+	// Get the commit object
+	if err != nil {
+		return nil, fmt.Errorf("git.CommitObject: %w", err)
+	}
+	f := false
+	// Create the BranchRef object
+	branchRef := &clients.BranchRef{
+		Name:      &ref.Name,
+		Protected: &f,
+	}
+
+	return branchRef, nil
+}
+
+func (c *Client) GetCreatedAt() (time.Time, error) {
```

Comment:
This could be first commit time, but I think clients.ErrUnsupportedFeature is fine for now. PR getting large

Reply:
Included a TODO to address this.